src/cli/self_update/windows.rs

```diff
@@ -995,15 +928,11 @@ mod tests {
         );
 
         let _guard = RegistryGuard::new(&USER_PATH);
-        let root = RegKey::predef(HKEY_CURRENT_USER);
-        let environment = root
-            .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
+
+        let environment = CURRENT_USER.create("Environment").unwrap();
+        environment
+            .set_hstring("PATH", &HSTRING::from_wide(&[0x12, 0x34]).unwrap())
```

Comment:
You should not set a `HSTRING` here, as the original code sets a binary (see https://docs.rs/windows-registry/latest/windows_registry/struct.Key.html#method.set_bytes).

Reply:
I met this error when passing row bytes into registry:
![屏幕截图 2024-07-05 105703](https://github.com/rust-lang/rustup/assets/146612054/717311f7-62bc-4cca-823c-11da8798608a)
I think it could be fixed, but so far I haven't found a solutions to fix this error yet. To test other tests, I temporarily ignored this test.

Reply:
I met this error when passing row bytes into registry:
![屏幕截图 2024-07-05 105703](https://github.com/rust-lang/rustup/assets/146612054/462e75fd-731f-4c87-b925-14873c4e0ba1)
I think it could be fixed, but so far I haven't found solutions to fix this error yet. To test other tests, I temporarily ignored this test.

Reply:
@InfyniteHeap Add the `Ok(None)` case back in https://github.com/rust-lang/rustup/pull/3896/files/9977fd34910921f0e2e5e1236346d36f5f3ccaaf#r1666169848 and you'll see this error gone.